Update to AniList Api v2

DaniGTA opened the pull request:
AniList API v1 is depracted and will not work anymore and give some errors in the log.
This is fixed with this version.

+ Fix some wrong results
+ Faster AniDBID Search
+ Name check is now async
+ MyAnimeList fix
+ Remove MyAnimeList login

DaniGTA committed:
- Faster AniDBID Search
- Hunter X Hunter (2011) == Hunter X Hunter (1999) is now false (fix ignore date)

DaniGTA committed:
y

DaniGTA committed:
Compability

DaniGTA committed:
Compability

DaniGTA committed:
Compability

DaniGTA committed:
Delete db.lock

DaniGTA committed:
Delete storage.ide

DaniGTA committed:
Delete db.lock

DaniGTA committed:
Fix github

DaniGTA committed:
Merge branch 'master' of https://github.com/MediaBrowser/Emby.Plugins.Anime

# Conflicts:
#	.vs/MediaBrowser.Plugins.Anime/v15/sqlite3/storage.ide

DaniGTA committed:
Update to AniList APIv2

+ Fixing some wrong results
+ (Beta)Added AniList Person infos

DaniGTA committed:
Revert "Update to AniList APIv2"

This reverts commit 57f5de0251176f8b779aedb880527c20544c77fd.

DaniGTA committed:
Revert "Revert "Update to AniList APIv2""

This reverts commit b3c30dd27e4e26040213c899ee15a47bbe99d463.

DaniGTA committed:
Deleting vs things

DaniGTA committed:
Raise version number

DaniGTA committed:
Add AniDB waiter.
You can set now a wait time to prevent a IP ban.

hapylestat commented on MediaBrowser.Plugins.Anime/Configuration/configPage.html:
```diff
@@ -43,13 +43,19 @@
                                 MyAnimeList Pw. -not rly needed
                             </label>
-                            <input id="chkMyAnimeList_API_Pw" name="chkMyAnimeList_API_Pw" type="password" value="" disabled/>
+                            <input id="chkMyAnimeList_API_Pw" name="chkMyAnimeList_API_Pw" type="password" value="" disabled />
```
if we would not need passwords in the future, corresponding configuration field could be removed

hapylestat commented on MediaBrowser.Plugins.Anime/Providers/AniList/api.cs:
```diff
@@ -0,0 +1,415 @@
+        private static IJsonSerializer _jsonSerializer;
+        public List<string> anime_search_names = new List<string>();
+        public List<string> anime_search_ids = new List<string>();
+        public string SearchLink = @"https://graphql.anilist.co/api/v2?query=
```
could be a private const

hapylestat commented on MediaBrowser.Plugins.Anime/Providers/AniList/api.cs:
```diff
@@ -0,0 +1,415 @@
+    }
+  }
+}&variables={ ""query"":""{0}"",""type"":""ANIME""}";
+        public string AniList_anime_link = @"https://graphql.anilist.co/api/v2?query=query($id: Int!, $type: MediaType) {
```
could be a protected internal const

hapylestat commented on MediaBrowser.Plugins.Anime/Providers/AniList/api.cs:
```diff
@@ -0,0 +1,415 @@
+    }
+        }
+    }&variables={ ""id"":""{0}"",""type"":""ANIME""}";
+        public string AniList_anime_char_link = @"https://graphql.anilist.co/api/v2?query=query($id: Int!, $type: MediaType, $page: Int = 1) {
```
could be a private const

hapylestat commented on MediaBrowser.Plugins.Anime/Providers/AniList/api.cs:
```diff
@@ -0,0 +1,415 @@
+                try
+                {
+
+                    if (await Task.Run(() => Equals_check.Compare_strings(media.title.romaji, title)))
```
cancellationToken is missing

hapylestat commented on MediaBrowser.Plugins.Anime/Providers/AniList/api.cs:
```diff
@@ -0,0 +1,415 @@
+                    {
+                        return media.id.ToString();
+                    }
+                    if (await Task.Run(() => Equals_check.Compare_strings(media.title.english, title)))
```
cancellationToken is missing

hapylestat commented on MediaBrowser.Plugins.Anime/Providers/AniList/api.cs:
```diff
@@ -0,0 +1,415 @@
+                    {
+                        return media.id.ToString();
+                    }
+                    //Disabled due to false result.
```
this could be just removed, if not needed ?

hapylestat commented on MediaBrowser.Plugins.Anime/Providers/AniList/api.cs:
```diff
@@ -0,0 +1,415 @@
+                try
+                {
+
+                    if (await Task.Run(() => Equals_check.Compare_strings(media.title.romaji, title)))
```
cancellationToken is missing

hapylestat commented on MediaBrowser.Plugins.Anime/Providers/AniList/api.cs:
```diff
@@ -0,0 +1,415 @@
+                    {
+                        result.Add(media.id.ToString());
+                    }
+                    if (await Task.Run(() => Equals_check.Compare_strings(media.title.english, title)))
```
cancellationToken is missing

hapylestat commented on MediaBrowser.Plugins.Anime/Providers/AniList/api.cs:
```diff
@@ -0,0 +1,415 @@
+                    {
+                        result.Add(media.id.ToString());
+                    }
+                    //Disabled due to false result.
```
this could be just removed, if not needed ?

hapylestat commented on MediaBrowser.Plugins.Anime/Providers/AniList/api.cs:
```diff
@@ -0,0 +1,415 @@
+        /// <returns></returns>
+        private string SelectName(RootObject WebContent, TitlePreferenceType preference, string language)
+        {
+            if (preference == TitlePreferenceType.Localized && language == "en")
```
Look more elegant? 
```
            switch (preference)
            {
                case TitlePreferenceType.Localized when language == "en":
                    return WebContent.data.Media.title.english;
                case TitlePreferenceType.Japanese:
                    return WebContent.data.Media.title.native;
            }
```

hapylestat commented on MediaBrowser.Plugins.Anime/Providers/AniList/api.cs:
```diff
@@ -0,0 +1,415 @@
+        /// </summary>
+        public async Task<string> One_line_regex(Regex regex, string match, int group = 1, int match_int = 0)
+        {
+            Regex _regex = regex;
```
unused variable

hapylestat commented:
Great patch! Taking in the account that AniList v1 API is deprecated and would be stopped soon.

DaniGTA committed:
Fix MyAnimeList
+ Removed some dead code in AniList
+ Anime name check is now async
+ Removed the MyAnimeList login function

-> If you want too use the MyAnimeList API you just need too remove the comments in the ConfigPage and compiled again.!!! Your username and PASSWORD will be saved in clear text !!!.

DaniGTA committed:
Fix some Regex for MyAnimeList
